### ISSUE TITLE
[Bug] DFlash misses Mamba checkpoints when accepted tokens cross a tracking boundary

### ISSUE THREAD

**Issue by simple-sun** [sgl-project/sglang#37817]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

We hit a cache consistency issue with Qwen3.6-35B-A3B using DFLASH and `mamba-radix-cache-strategy=extra_buffer`.

The model has both full-attention KV cache and linear-attention state. The linear state is checkpointed at fixed boundaries (256 tokens in our setup). If a DFlash verify step crosses one of these boundaries and then rolls back rejected tokens, the active linear state is restored to the last accepted token, but the state at the checkpoint boundary is not copied to the tracking buffer.

The problem appears to be in `DFlashWorkerV2._update_target_mamba_state_after_verify`:

```python
to_track_mask = (
    seq_lens_pre_verify // mamba_track_interval
    != batch.seq_lens // mamba_track_interval
)
```

`seq_lens_pre_verify` is cloned from `batch.seq_lens`, and `batch.seq_lens` is not advanced before this check. Only `batch.seq_lens_cpu` is temporarily changed for backend planning. As a result, the two sides above use the same length and no boundary is detected.

For example, with an interval of 256:

- length before verify: 255
- verify block size: 16
- committed tokens after verification: 1
- resulting committed length: 256

Step 0 should be saved as the state checkpoint at length 256, but `mamba_steps_to_track` is set to -1.

The scheduler-side result handling does detect the accepted range crossing the boundary and updates `mamba_last_track_seqlen` and the ping-pong buffer bookkeeping. This can leave a radix-cache entry whose full-attention KV corresponds to length 256 while its linear-attention state is stale.

The shared speculative path in `spec_utils._verify_commit_step_indices` calculates the post-verify length from the accepted length:

```python
seq_lens_post_verify = batch.seq_lens + accept_lens
```

The DFlash path likely needs the equivalent calculation based on `seq_lens_pre_verify + commit_lens`.

Related: #35821 and #37326 cover nearby accepted-state tracking problems, but do not appear to cover this DFlash-specific calculation.

We found this during a joint investigation. The fix is ready for review in #37818.

### Reproduction

The boundary condition can be reduced to:

```python
import torch

interval = 256
pre = torch.tensor([255])
batch_seq_lens = pre.clone()
commit_lens = torch.tensor([1])

current = pre // interval != batch_seq_lens // interval
expected = pre // interval != (pre + commit_lens) // interval

assert current.item() is False
assert expected.item() is True
```

Configuration under test:

```text
Model: Qwen/Qwen3.6-35B-A3B
Speculative algorithm: DFLASH
Mamba radix cache strategy: extra_buffer
Mamba track interval: 256
```

Expected behavior for representative boundary cases:

- pre=255, commit=1: save step 0 at 256
- pre=254, commit=2: save step 1 at 256
- pre=250, commit=3: do not save a checkpoint
- pre=256, commit=1: do not save 256 again

### Environment

Repository: sgl-project/sglang
Branch: main
Commit: f5bed255c0d855a1ab98bd3a7df7297a33dca5e3
Model: Qwen/Qwen3.6-35B-A3B

Validation is complete. See #37818 for the fix, regression test, and GPU validation results.

**Comment by simple-sun** [comment]
We’ve fixed this issue in #37818. The PR is ready for review.
